source/MaterialXGenOsl/OslSyntax.cpp

```diff
@@ -246,6 +246,68 @@ class OSLMatrix3TypeSyntax : public AggregateTypeSyntax
     }
 };
 
+class FileFormatSyntax : public AggregateTypeSyntax
+{
+  public:
+    FileFormatSyntax(const string& name, const string& defaultValue, const string& uniformDefaultValue,
+                     const string& typeAlias = EMPTY_STRING, const string& typeDefinition = EMPTY_STRING,
+                     const StringVec& members = EMPTY_MEMBERS) :
+        AggregateTypeSyntax(name, defaultValue, uniformDefaultValue, typeAlias, typeDefinition, members)
+    {}
+    
+    string getValue(const ShaderPort* port, bool uniform) const override
+    {
+        if (!port)
+        {
+            return EMPTY_STRING;
+        }
+
+        StringVec values;
+        values.push_back(port->getValue() ? port->getValue()->getValueString() : EMPTY_STRING);
+        values.push_back(port->getColorspace());
+        return getValue(values, uniform);
```

Comment:
I'm not sure you need the extra getValue(StringVec) utility method. It would be quite simple and much more efficient to just construct and return the value string directly here. Avoiding the vector construction and std::stringstream usage inside the utility.

Reply:
I'll look into streamlining this.